File: app/src/main/java/com/example/xc_m6_java/activity/LanguageActivity.java
package com.example.xc_m6_java.activity;

import android.content.Context;
import android.content.res.Configuration;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.xc_m6_java.MyApplication;
import com.example.xc_m6_java.R;
import com.example.xc_m6_java.databinding.ActivityLanguageBinding;
import com.example.xc_m6_java.manager.LocaleManager;

import java.util.Locale;

public class LanguageActivity extends AppCompatActivity {

    ActivityLanguageBinding binding;
    Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityLanguageBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        initViews();
    }

    private void initViews() {
        context = this;

        binding.bEnglish.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                MyApplication.instance.localeManager.setNewLocale(context, LocaleManager.LANGUAGE_ENGLISH);
                finish();
            }
        });

        binding.bRussian.setOnClickListener(new View.OnClickListener() {
            @Override

            public void onClick(View view) {
                MyApplication.instance.localeManager.setNewLocale(context,LocaleManager.LANGUAGE_RUSSIAN);
                finish();
            }
        });

        binding.bUzbek.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                MyApplication.instance.localeManager.setNewLocale(context,LocaleManager.LANGUAGE_UZBEK);
                finish();
            }
        });
        // one = 1
        String one = getResources().getQuantityString(R.plurals.my_cats,1,1);
        // few = 2~3
        String few = getResources().getQuantityString(R.plurals.my_cats,2,3);
        // many = 5~
        String many = getResources().getQuantityString(R.plurals.my_cats,5,10);

        Log.d("LanguageActivity one",one);
        Log.d("LanguageActivity few",few);
        Log.d("LanguageActivity many",many);
    }

    private void setLocale(String lan) {
        Locale locale = new Locale(lan);
        Locale.setDefault(locale);
        Configuration config = new Configuration();
        config.locale = locale;
        getBaseContext().getResources().updateConfiguration(config,getBaseContext().getResources().getDisplayMetrics());
        finish();
    }
}

File: app/src/main/java/com/example/xc_m6/activity/LanguageActivity.kt
package com.example.xc_m6.activity

import android.content.Context
import android.content.res.Configuration
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.xc_m6.MyApplication
import com.example.xc_m6.R
import com.example.xc_m6.manager.LocaleManager
import kotlinx.android.synthetic.main.activity_language.*
import kotlinx.android.synthetic.main.activity_preference.*
import java.util.*

class LanguageActivity : AppCompatActivity() {

    lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_language)
        initViews()
    }

    private fun initViews() {
        context = this
        b_english.setOnClickListener {
            MyApplication.localeManger!!.setNewLocale(context,LocaleManager.LANGUAGE_ENGLISH)
            finish()
        }

        b_russian.setOnClickListener {
            MyApplication.localeManger!!.setNewLocale(context,LocaleManager.LANGUAGE_RUSSIAN)
            finish()
        }

        b_uzbek.setOnClickListener {
            MyApplication.localeManger!!.setNewLocale(context,LocaleManager.LANGUAGE_UZBEK)
            finish()
        }

        // one = 1
        val one = resources.getQuantityString(R.plurals.my_cats, 1, 1)
        // few = 2~4
        val few = resources.getQuantityString(R.plurals.my_cats, 2, 3)
        // many = 5~
        val many = resources.getQuantityString(R.plurals.my_cats, 5, 10)

        Log.d("LanguageActivity one", one)
        Log.d("LanguageActivity few", few)
        Log.d("LanguageActivity many", many)
    }

    private fun setLocale(lan: String) {
        val locale = Locale(lan)
        Locale.setDefault(locale)
        val config = Configuration()
        config.locale = locale
        baseContext.resources.updateConfiguration(config, baseContext.resources.displayMetrics)
        finish()
    }
}